scheduler/scheduler.py

```diff
@@ -2643,17 +2643,20 @@ def _init_job_callback(self, job_id):
             scale_factor = self._jobs[job_id].scale_factor
             remaining_steps = self._get_remaining_steps(job_id)
             remaining_steps = int(math.ceil(remaining_steps / scale_factor))
+            current_time = self.get_current_timestamp()
+            current_round_end_time = \
+                self._current_round_start_time + self._time_per_iteration
+            remaining_time_in_current_round = \
+                current_round_end_time - current_time
 
             # Add additional time to the lease if the job was dispatched early.
-            extra_time = 0
             if (self._next_worker_assignments is not None and
                 next_job_combination is not None):
-                current_time = self.get_current_timestamp()
-                elapsed_time_in_round = \
-                    current_time - self._current_round_start_time
-                extra_time += self._time_per_iteration - elapsed_time_in_round
-                extra_time = max(extra_time, 0)
-            return (remaining_steps, self._time_per_iteration, extra_time)
+                extra_time = remaining_time_in_current_round
+                return (remaining_steps, self._time_per_iteration,
```

Comment:
Add a comment somewhere what the tuple you return means

Reply:
Done